src/core/agent/services/connectionService.ts

```diff
@@ -395,7 +390,20 @@ class ConnectionService extends AgentService {
     } else if (!operation.done) {
       throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
     }
-    const oobi = { ...operation, alias };
+
+    const connectionId =
+      operation.done && operation.response
+        ? operation.response.i
+        : new URL(url).pathname.split("/oobi/").pop()!.split("/")[0];
+
+    const keriaConnection = await this.props.signifyClient
+      .contacts()
+      .update(connectionId, {
+        alias,
+        groupCreationId: groupId,
+        createdAt: new Date().toISOString(),
```

Comment:
Might be a little clearer when the other PR merges, but we should probably avoid calling `contacts().update()` if the contact already exists. Just in case we re-resolve by accident and overwrite the creation timestamp. Maybe in the long running operation tracker we can check if it already exists and skip if it does

Reply:
So, as I understand, we just create a contact when the long running operation and scan QR
In the long running operation, we need to check contact already exists to avoid call update again, isn't it

Reply:
@Sotatek-TungNguyen2a this still isnt resolved